### PR TITLE
Add scikit-build-core

### PR DESCRIPTION
**Pull request by henryiii**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/scikit-build-core/meta.yaml
Comment (henryiii) [inline recipes/scikit-build-core/meta.yaml:10]:
```suggestion
  sha256: d6fb6f71447e512cf0e2b998d644669997c820ddd09ed10d70ab9482a5d34b5d
```

File: recipes/scikit-build-core/meta.yaml
Comment (henryiii) [inline recipes/scikit-build-core/meta.yaml:2]:
```suggestion
{% set version = "0.1.0b1" %}
```

**Comment by henryiii** [comment]
@conda-forge/help-python, this is ready!

File: recipes/scikit-build-core/meta.yaml
Comment (BastianZim) [inline recipes/scikit-build-core/meta.yaml:2]: This should probably target pre-release builds: https://conda-forge.org/docs/maintainer/knowledge_base.html#pre-release-builds

File: recipes/scikit-build-core/meta.yaml
Comment (BastianZim) [inline recipes/scikit-build-core/meta.yaml:25]: This and some others are not listed upstream. As you're both upstream and maintainer and familiar with conda-forge, I assume that they are needed?

File: recipes/scikit-build-core/meta.yaml
Comment (BastianZim) [inline recipes/scikit-build-core/meta.yaml:48]: Are you happy just testing this on Linux? Otherwise, you can add multiple noarch platforms.

**Review by BastianZim — CHANGES_REQUESTED** [review]

File: recipes/scikit-build-core/meta.yaml
Comment (henryiii) [reply recipes/scikit-build-core/meta.yaml:2]: There is no non-prerelease build, so I was thinking this is best. And the main reason it's still pre-release is so I can verify it works on conda-forge first.

File: recipes/scikit-build-core/meta.yaml
Comment (henryiii) [reply recipes/scikit-build-core/meta.yaml:48]: I'd like to do that, I haven't seen that yet.

File: recipes/scikit-build-core/meta.yaml
Comment (henryiii) [reply recipes/scikit-build-core/meta.yaml:25]: They are added by the `get_requires_for_build_wheel` (and sdist) hooks (and some of them are also in the pyproject extra, minus cmake & ninja) The one extra requirement is `make` - sadely, Conda-build injects `CMAKE_GENERATOR="Unix Makefiles"` even if make is not installed (and Ninja is)!

File: recipes/scikit-build-core/meta.yaml
Comment (jcfr) [inline recipes/scikit-build-core/meta.yaml:70]:
```suggestion
    - henryiii
    - jcfr
    - thewtex
```

Adding @thewtex and myself (@jcfr) to the list of maintainer.

File: recipes/scikit-build-core/meta.yaml
Comment (BastianZim) [reply recipes/scikit-build-core/meta.yaml:2]: Ahh ok. In that case, you should still be able to make a prerelease build and test it by adding the respective flags but let me ask core, though, to make sure. @ocefpaf, what do you think?

File: recipes/scikit-build-core/meta.yaml
Comment (BastianZim) [reply recipes/scikit-build-core/meta.yaml:25]: Ahh ok makes sense.

File: recipes/scikit-build-core/meta.yaml
Comment (BastianZim) [reply recipes/scikit-build-core/meta.yaml:48]: Just add a conda-forge.yml here with the following keys:
noarch_platforms:
  - linux_64
  - win_64
  - osx_64

Thinking of it, this does require, though, that only one package is uploaded. Let me ask in the chat and maybe hold off for now.

Ref: https://conda-forge.org/docs/maintainer/conda_forge_yml.html#noarch-platforms

File: recipes/scikit-build-core/meta.yaml
Comment (BastianZim) [reply recipes/scikit-build-core/meta.yaml:70]: @thewtex Are you happy to be added as a maintainer here?

Comment (thewtex) [reply recipes/scikit-build-core/meta.yaml:70]: Yes, thanks!

File: recipes/scikit-build-core/meta.yaml
Comment (BastianZim) [reply recipes/scikit-build-core/meta.yaml:48]: OK seems to be more difficult as then three packages would be uploaded, which needs to be restricted again.

Comment (henryiii) [reply recipes/scikit-build-core/meta.yaml:48]: That’s fine. The best test will be preparing awkward-cpp on all platforms. I’ll be releasing an rc in the next hour or two, I can update this (some recent testing improvements should help here).

**Comment by henryiii** [comment]
@conda-forge/help-python, this is ready (again)! Using the just released non-prerelease release now.

File: recipes/scikit-build-core/meta.yaml
Comment (henryiii) [reply recipes/scikit-build-core/meta.yaml]: It's 0.1.0 now!

**Pull request merged**
